src/core/logging/logger.hpp

```diff
@@ -343,42 +350,96 @@
-#define log_and_throw_io_failure(message)                             \
-  do {                                                                \
-    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {    \
-      logstream(LOG_ERROR) << (message) << std::endl;                 \
-      throw(turi::error::io_error(message, std::error_code()));      \
-    };                                                                \
-    throw_error();                                                    \
-  } while(0)
+#define log_and_throw_io_failure(message)                       \
+  do {                                                          \
+    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {           \
+      logstream(LOG_ERROR) << (message) << std::endl;           \
+      throw(turi::error::io_error(message, std::error_code())); \
+    };                                                          \
+    throw_error();                                              \
+  } while (0)
 #else
-#define log_and_throw_io_failure(message)                             \
-  do {                                                                \
-    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {    \
-      logstream(LOG_ERROR) << (message) << std::endl;                 \
-      throw(turi::error::io_error(message));                         \
-    };                                                                \
-    throw_error();                                                    \
-  } while(0)
+#define log_and_throw_io_failure(message)             \
+  do {                                                \
+    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR { \
+      logstream(LOG_ERROR) << (message) << std::endl; \
+      throw(turi::error::io_error(message));          \
+    };                                                \
+    throw_error();                                    \
+  } while (0)
 #endif
 
-#define log_and_throw_current_io_failure()                            \
-  do {                                                                \
-    auto error_code = errno;                                          \
-    std::string error_message = std::strerror(error_code);            \
-    errno = 0; /* clear errno */                                      \
-    log_and_throw_io_failure(error_message);                          \
-  } while(0)
+#else // debug mode
 
-#define log_func_entry()                                       \
-  do {                                                         \
-    logstream(LOG_INFO) << "Function entry" << std::endl;      \
-  } while(0)
+#define log_and_throw(message)                                          \
+  do {                                                                  \
+    auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {                   \
+      std::stringstream _turi_ss;                                       \
+      _turi_ss << (message) << ". " << __func__ << " from " << __FILE__ \
```

Comment:
Just out of curiosity, why do we only want to include filename and line numbers in debug builds, and not in release too? This message is only ever going to a debug log anyway.

Reply:
I think we mainly work on debug mode and it's reasonable to include those detailed bug info. But for the end-users, we don't want too much irrelevant info to distract them. I personally prefer to include in release mode so that the end-user can simply copy and paste everything to report.

Reply:
Regular users don't see the debug log at all, do they? I thought it just goes to disk.

Reply:
The log and exception are different. I doubt they would like to read the log since it's very troublesome to do so (you have to find log file location and then copy-paste the corresponding logs). Most cases I've seen so far are people just copy-paste what's reported in the interactive shell.

if we include src info in exception, we can debug fairly faster within the shell instead of tailing the log.